docarray/typing/url/video_url.py

```diff
@@ -0,0 +1,180 @@
+from typing import TYPE_CHECKING, Any, Tuple, Type, TypeVar, Union
+
+import numpy as np
+from pydantic.tools import parse_obj_as
+
+from docarray.typing import AudioNdArray, NdArray
+from docarray.typing.tensor.video import VideoNdArray
+from docarray.typing.url.any_url import AnyUrl
+
+if TYPE_CHECKING:
+    from pydantic import BaseConfig
+    from pydantic.fields import ModelField
+
+    from docarray.proto import NodeProto
+
+T = TypeVar('T', bound='VideoUrl')
+
+VIDEO_FILE_FORMATS = ['mp4']
+
+
+class VideoUrl(AnyUrl):
+    """
+    URL to a .wav file.
+    Can be remote (web) URL, or a local file path.
+    """
+
+    def _to_node_protobuf(self: T) -> 'NodeProto':
+        """Convert Document into a NodeProto protobuf message. This function should
+        be called when the Document is nested into another Document that needs to
+        be converted into a protobuf
+        :return: the nested item protobuf message
+        """
+        from docarray.proto import NodeProto
+
+        return NodeProto(video_url=str(self))
+
+    @classmethod
+    def validate(
+        cls: Type[T],
+        value: Union[T, np.ndarray, Any],
+        field: 'ModelField',
+        config: 'BaseConfig',
+    ) -> T:
+        url = super().validate(value, field, config)
+        has_video_extension = any(ext in url for ext in VIDEO_FILE_FORMATS)
+        if not has_video_extension:
+            raise ValueError(
+                f'Video URL must have one of the following extensions:'
+                f'{VIDEO_FILE_FORMATS}'
+            )
+        return cls(str(url), scheme=None)
+
+    def _load(
+        self: T, skip_type: str, **kwargs
+    ) -> Tuple[AudioNdArray, VideoNdArray, NdArray]:
+        """
+        Load the data from the url into a Tuple of AudioNdArray, VideoNdArray and
+        NdArray.
+
+        :param skip_type: determines what video frames to discard.
```

Comment:
can u list the supported string here ? Could be even better to have an Enum

Reply:
okay maybe not needed to have an Enum as you pass it to another function directly